conjur/api/api.py

```diff
@@ -105,6 +107,23 @@ def login(self, login_id: str = None, password: str = None) -> str:
 
         return self.api_key
 
+    def login_ldap(self, login_id: str = None, password: str = None) -> str:
+        """
+        This method uses the LDAP auth login id (username) and password
+        to retrieve an api key from the server that can be later used to
+        retrieve short-lived api tokens.
+        """
+        if not login_id or not password:
+            raise MissingRequiredParameterException("Missing parameters in login invocation!")
+
+        logging.debug("Logging in to %s...", self._url)
+        self.api_key = invoke_endpoint(HttpVerb.GET, ConjurEndpoint.LOGIN_LDAP,
+                                       self._default_params, auth=(login_id, password),
+                                       ssl_verify=self._ssl_verify).text
+        self.login_id = login_id
+
+        return self.api_key
```

Comment:
This code snippet seems to be repeated above under the function `login(...)`. Because of this, it would be best to extract the common logic into a separate function where we can feed `ConjurEndpoint.LOGIN_LDAP` and `ConjurEndpoint.LOGIN` as parameters. All we have to do is call the function `login` with the different parameters. 

For example:
```
def login(self, login_id: str = None, password: str = None, authentication_type=ConjurEndpoint.LOGIN) -> str:
...
  logging.debug("Logging in to %s...", self._url)
  self.api_key = invoke_endpoint(HttpVerb.GET, login_type,
                          self._default_params, auth=(authentication_type, password),
                           ssl_verify=self._ssl_verify).text
....
```

Reply:
might also be worth adding to the debug message which authentication method we are using (ldap or basic authn)